File: examples/rust/constructs/array_mutations.rs
// Ported from examples/lumen/constructs/array_mutations.lm by scripts/port_examples.py; edit the Lumen original, not this file.
fn main() {
    println!("=== Array Features Test ===");
    let arr = [10, 20, 30];
    print!("Array: ");
    println!("{}", arr);
    print!("arr[0] = ");
    println!("{}", arr[0]);
    let arr2 = [1, 2, 3];
    arr2[1] = 999;
    print!("After arr2[1]=999: ");
    println!("{}", arr2);
    let arr3 = [];
    arr3.push(100);
    arr3.push(200);
    print!("After push: ");
    println!("{}", arr3);
    println!("Done!");
}

File: examples/python/constructs/array_mutations.py
import sys
# Ported from examples/lumen/constructs/array_mutations.lm by scripts/port_examples.py; edit the Lumen original, not this file.
print("=== Array Features Test ===")
arr = [10, 20, 30]
sys.stdout.write("Array: ")
print(arr)
sys.stdout.write("arr[0] = ")
print(arr[0])
arr2 = [1, 2, 3]
arr2[1] = 999
sys.stdout.write("After arr2[1]=999: ")
print(arr2)
arr3 = []
arr3.append(100)
arr3.append(200)
sys.stdout.write("After push: ")
print(arr3)
print("Done!")
